json_encoder_test.go

```diff
@@ -66,6 +88,7 @@ func TestJSONAddString(t *testing.T) {
 
 func TestJSONAddBool(t *testing.T) {
 	withJSONEncoder(func(enc *jsonEncoder) {
+		enc.AddString("foo", "bar")
```

Comment:
Ah OK, by testing that there's a comma separating them, we're ignoring the case where there's no comma separator right now.

Can we have a test helper that's intended for testing adding a field that will run multiple cases:
 - Don't add a field before, make sure there's no comma
 - Add a field before (and after), make sure there's a comma